File: src/software/HotPi/Reflow/Ui/Angular/src/app/oven-temperature-sampled-observable/oven-temperature-sampled.ts
import { Temperature } from './temperature';

export class OvenTemperatureSampled {
  constructor(private _timestamp: Date, private _temperature: Temperature) {
  }

  public get timestamp(): Date {
    return this._timestamp;
  }

  public get temperature(): Temperature {
    return this._temperature;
  }
}

File: src/software/HotPi/Reflow/Oven/OvenTemperatureSampled.cs
using System;
using System.Diagnostics.CodeAnalysis;

namespace Restall.HotPi.Reflow.Oven
{
	public class OvenTemperatureSampled
	{
		public OvenTemperatureSampled(DateTime timestamp, Temperature temperature)
		{
			this.Timestamp = timestamp;
			this.Temperature = temperature;
		}

		[SuppressMessage("ReSharper", "AutoPropertyCanBeMadeGetOnly.Local", Justification = CodeAnalysisJustification.ForSerialisation)]
		public DateTime Timestamp { get; private set; }

		[SuppressMessage("ReSharper", "AutoPropertyCanBeMadeGetOnly.Local", Justification = CodeAnalysisJustification.ForSerialisation)]
		public Temperature Temperature { get; private set; }
	}
}
